### ISSUE TITLE
rpi image has missing chrony user/group

### ISSUE THREAD

**Issue by BlackLotus** [void-linux/void-mklive#4]
The rpi2 image (didn't test the others) is missing the chrony user+group.
Had to create the user myself.

Alternatively I could just do
> xbps-install -Su chrony

After setting the time manually

**Comment by amak79** [comment]
I just built an `rpi` image and this is still an issue. I ran `xbps-reconfigure -f chrony` to create the user/group. I also noticed that the ntpd service is no longer enabled by default. This requires the date/time to bet set manually otherwise xbps-install will fail with certificate verification errors.

**Comment by amak79** [comment]
Further testing revealed that both the `rpi` (armv6l) and `rpi2` (armv7l) images are affected, while the `rpi3` (aarch64) image is not. The `rpi` and `rpi2` images are also missing the service links for `dhcpcd`, `ntpd` and `sshd`.

**Comment by jacereda** [comment]
Faced this same issue on a fresh install today on `rpi`.

**Comment by xtraeme** [comment]
is this a problem in the host? (the host does not have correct time/date?)

**Comment by amak79** [comment]
Yes. Without the `chrony` user and `ntpd` service link, chrony is unable to start during boot. This results in incorrect date/time (Unix epoch) for the host/device.

**Comment by xtraeme** [comment]
But it's expected the host has proper date/time settings. checking for SSL certs really need correct time/date. How could void-mklive solve this for you?

**Comment by xtraeme** [comment]
You are the user executing this script (and sequence of commands) that will generate the resulting image... if the host is ready for this operation why should it workaround it in its code?

**Comment by xtraeme** [comment]
The code could be smart enough to abort if any condition is not met.

**Comment by amak79** [comment]
I think I misunderstood your initial question. Are you asking if the host building the image has the correct date/time? If so, then yes it does.

**Comment by xtraeme** [comment]
I just built a pinebookpro (and musl) via mkrootfs/mkplatforms... how's this a problem?

**Comment by xtraeme** [comment]
I'd need more info of the host next time you report an issue. void-mklive depends on external artifacts to produce a result (host environment, sw version, etc).

**Comment by amak79** [comment]
> I just built a pinebookpro image (musl as well) via mkrootfs/mkplatforms... how's this a problem?

It's a problem because the generated `rpi` and `rpi2` images come with chrony installed, but it's unable to run because the `chrony` user and ntpd service link is missing. The Raspberry Pi doesn't have an RTC so a properly configured NTP client by default is necessary.

The `rpi3` image doesn't have this issue because the `chrony` user and `ntpd` service links are present in the image. The correct date/time is set during every boot by default. I would expect the `rpi` and `rpi2` images to behave in the same way.

**Comment by xtraeme** [comment]
Hmm I'll try.

**Comment by xtraeme** [comment]
```
Running tar -cp --posix --xattrs -C /tmp/tmp.qTX1FNAyTQ . | xz -T0 -9 > void-rpi3-musl-PLATFORMFS-20200220.tar.xz
Successfully created void-rpi3-musl-PLATFORMFS-20200220.tar.xz (rpi3-musl)
```
not a single error... what gives?

**Comment by amak79** [comment]
I see the following error when running `./mkrootfs.sh armv6l`:
```
95 downloaded, 95 installed, 0 updated, 95 configured, 0 removed.
Reconfiguring packages for armv6l ...
Running xbps-reconfigure --rootdir /tmp/tmp.bMTR3W7cpr base-files
base-files: configuring ...
ERROR: base-files: [configure] INSTALL script failed to execute the post ACTION: Exec format error
Failed to reconfigure `base-files': Exec format error
base-files: configuring ...
```
The rootfs is successfully created :
```
Running tar -cp --posix --xattrs -C /tmp/tmp.kGB4BAkjBg . | xz -T0 -9 > void-rpi-PLATFORMFS-20200221.tar.xz 
Successfully created void-rpi-PLATFORMFS-20200221.tar.xz (rpi)
```
The resulting `rpi` platformfs image is still missing the `chrony` user and `ntpd` service link. The only services that are present in `/etc/runit/runsvdir/default` are `agetty-tty{1-6}` and `udevd`. Compare that to the `rpi3` platform image which has `agetty-tty{1-6}`, `dhcpcd`, `ntpd`, `sshd`,  and `udevd`.

**Comment by xtraeme** [comment]
Hmm not sure, because next invokation runs ok with qemu-user-static just fine....

**Comment by amak79** [comment]
I was able to fix this issue by forcing package configure in `mkplatformfs.sh.in` line 169 :
```
run_cmd_chroot "$ROOTFS" "xbps-reconfigure -af"
```
It still doesn't explain why the `rpi3` image didn't have this issue.

**Comment by amak79** [comment]
I've noticed that platformfs images that have a 32-bit rootfs (`rpi`, `rpi2`, `beaglebone`, `cubieboard2`, `usbarmory`) are affected, while those that have a 64-bit rootfs (`odroid-c2`, `rpi3`) aren't.

**Comment by xtraeme** [comment]
reported fixed by #120 

Thanks
